### PR TITLE
feat: Add vote extensions

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Adds Slinky's vote extensions to the chain.

Also updated the docker-compose and config handling so that `make localnet-start` successfully pulls prices from the slinky container and posts them to vote extensions. 

I did not hook anything up in the proposal handlers.

### CODE REVIEW

File: protocol/app/app.go [hunk 1]
```diff
@@ -1340,6 +1398,22 @@ func New(
...
+		compression.NewCompressionVoteExtensionCodec(
+			compression.NewDefaultVoteExtensionCodec(),
+			compression.NewZLibCompressor(),
+		),
+		app.PreBlocker,
+		oracleMetrics,
+	)
+	app.SetExtendVoteHandler(voteExtensionsHandler.ExtendVoteHandler())
+	app.SetVerifyVoteExtensionHandler(voteExtensionsHandler.VerifyVoteExtensionHandler())
+
```
Comment (g3ndo9kar9) [inline protocol/app/app.go]: nit: we shld probs over-ride the app's Close() fn. to close the oracle-client / prom-server

File: protocol/app/app.go [hunk 2]
```diff
@@ -1287,6 +1301,50 @@ func New(
 	app.MountTransientStores(tkeys)
 	app.MountMemoryStores(memKeys)
 
+	// Slinky setup
```
Comment (g3ndo9kar9) [inline protocol/app/app.go]: We shld ensure that full-nodes aren't running the oracle? Is there a config parameter to control this?

**Review by g3ndo9kar9 — COMMENTED** [review]
lgtm, questions. We should check to see if the prices that we're extending votes w/ are in alignment w/ how  dydx gets prices in PrepareProposal? Currently they use their index-cache + the smoothed historical prices.

**Review by g3ndo9kar9 — APPROVED** [review]
lgtm

**Comment by Eric-Warehime** [comment]
> lgtm, questions. We should check to see if the prices that we're extending votes w/ are in alignment w/ how dydx gets prices in PrepareProposal? Currently they use their index-cache + the smoothed historical prices.

Yeah we probably want to add that logic in ExtendVote. Will add in a separate PR

File: protocol/app/app.go [hunk 3]
```diff
@@ -1287,6 +1300,11 @@ func New(
 	app.MountTransientStores(tkeys)
 	app.MountMemoryStores(memKeys)
 
+	// if the node is a NonValidatingFullNode, we don't need to run any of the oracle code
+	if !appFlags.NonValidatingFullNode {
```
Comment (g3ndo9kar9) [inline protocol/app/app.go:1304]: kek

File: protocol/app/app.go [hunk 4]
```diff
@@ -1660,6 +1739,12 @@ func (app *App) setAnteHandler(txConfig client.TxConfig) {
 // Close invokes an ordered shutdown of routines.
 func (app *App) Close() error {
```
Comment (g3ndo9kar9) [inline protocol/app/app.go:1740]: kek

**Pull request merged**